app/src/main/java/nz/org/cacophony/birdmonitor/PermissionsHelper.java

```diff
@@ -31,54 +31,51 @@ public void checkAndRequestPermissions(Activity activity, String... permissions)
     }
 
     public void onRequestPermissionsResult(Activity activity, int requestCode, String permissions[], int[] grantResults) {
-        switch (requestCode) {
-            case REQUEST_ID_MULTIPLE_PERMISSIONS: {
-                Map<String, Integer> perms = new HashMap<>();
-
-                for (String permission : permissions) {
-                    perms.put(permission, PackageManager.PERMISSION_GRANTED);
-                }
 
-                if (grantResults.length > 0) {
-                    for (int i = 0; i < permissions.length; i++)
-                        perms.put(permissions[i], grantResults[i]);
+        ArrayList<String> deniedPermissions = new ArrayList<>();
 
-                    boolean allPermissionsGranted = true;
-                    for (String permission1 : permissions) {
-                        allPermissionsGranted = allPermissionsGranted && (perms.get(permission1) == PackageManager.PERMISSION_GRANTED);
-                    }
+        for (int i = 0; i < grantResults.length; i++) {
+            if (grantResults[i] != PackageManager.PERMISSION_GRANTED) {
+                String deniedPermission = permissions[i];
+                try {
+                    // Each permission seems to start with 'android.permission.' so have removed these parts to make message nicer
+                    String[] parts = deniedPermission.split("\\.");
+                    deniedPermission = parts[parts.length - 1];
+                } catch (Exception ex) {
```

Comment:
why is this catch required? Even if there's no dot in the string parts will still have a length of at least 1.

Reply:
I agree - Done